static/app/components/arithmeticInput/parser.tsx

```diff
@@ -82,15 +111,18 @@ function flatten(remaining: Array<Operation>): Operation {
   return term;
 }
 
-export function parseArithmetic(query: string): {
-  result: Term;
+type parseResult = {
+  result: Expression;
   error: string | undefined;
-} {
+  tc: TokenConverter;
+};
+
+export function parseArithmetic(query: string): parseResult {
   const tc = new TokenConverter();
   try {
     const result = grammar.parse(query, {tc});
-    return {result, error: tc.errors[0]};
+    return {result, error: tc.errors[0], tc};
```

Comment:
Nit: Not sure if it's still necessary to lift `error` to a top-level item in the return value here since you're also returning `tc`.

Reply:
Still need to since on exceptions the error becomes the exception message instead